### DISCUSSION TITLE
Add support for twitter legacy oauth 1.0A in v5

### DISCUSSION THREAD

**Discussion by svarup** [nextauthjs/next-auth#11711] [Ideas]
### Goals

As in Twitter OAuth 2.0, there is no way to get the user's email address. Twitter does not provide any [Scope ](https://developer.x.com/en/docs/x-api/users/lookup/api-reference/get-users-me) to get an authenticated user's email address. Even if you check "Request email from users" in the Twitter App you won't get the email address. But if you use OAuth 1.0A you can request email using "include_email=true" scope.

### Background

Currently, in [v5 ](https://github.com/nextauthjs/next-auth/blob/next-auth%405.0.0-beta.20/packages/core/src/providers/twitter.ts) there is only an OAuth 2.0 way in Twitter but we can't get email address. Please add OAuth 1.0A support that we have in [v4](https://github.com/nextauthjs/next-auth/blob/v4.2.1/src/providers/twitter.ts)

### Proposal

Please add legacy Twitter OAuth 1.0A provider in V5

**Comment by matt-roz** [comment]
I had to painfully come to the conclusion that requiring `email` for them user locks one into Twitter OAuth 1.0A, which is no longer supported by v5. 

This unfortunately means v4 users that employ this restriction (requiring `email`) cannot migrate to v5.

**Comment by steipete** [comment]
Running into the same issue. my flow is OAuth 2 -> OAuth 1 -> App. Not great, but better than letting the user manually enter their email and then waiting for them click verify. 

Edit: I was wrong! There is `confirmed_email` now: https://devcommunity.x.com/t/announcing-support-for-email-address-retrieval-with-oauth-2-0-in-the-x-api-v2/240555/4

**Comment by IDM-desaive** [comment]
I still think there's a need for the OAuth 1.0A provider.

We currently have an X application in "standalone" project. We have an existing application relying on HybridAuth using OAuth 1.0A successfully.

As we plan to rewrite this web application, I tried to use Next-auth with it. As Next-Auth removed support for OAuth 1.0A I could not get it to work without switching the X app from "standalone" to "pay per use", which we definitely don't want to do.

**Reply by IDM-desaive** [reply]
Also, in the migration guide (https://authjs.dev/getting-started/migrating-to-v5), it's written that OAuth 1.0 has been deprecated, not that it has been removed.
However, I can not find any documentation on how I could write an OAuth 1.0 provider in next-auth 5.
